test-server/test-echo.c

```diff
@@ -33,6 +33,9 @@
 #include <syslog.h>
 #include <sys/time.h>
 #include <unistd.h>
+#else
+#include "../win32port/win32helpers/gettimeofday.h"
```

Comment:
Basically none of the changes in this file have any relation to MinGW support, so remove them.